Bigtable/tests/Unit/TableTest.php

```diff
@@ -198,6 +198,47 @@ public function testMutateRowsApiExceptionInMutateRows()
         $this->table->mutateRows($this->rowMutations);
     }
 
+    public function testMutateRowsApiExceptionThrowsErrorInfo()
+    {
+        $exMessage = 'unauthenticated';
+        $apiException = new ApiException(
+            $exMessage,
+            Code::UNAUTHENTICATED,
+            'unauthenticated',
+            [
+                'metadata' => [
+                    [
+                        'reason' => 'some failure reason',
+                        'domain' => 'some failure domain',
+                        'metadata' => [
+                            'service' => 'bigtable.googleapis.com',
+                            'consumer' => 'projects\/my-project',
+                        ],
+                    ],
+                ],
+            ]
+        );
+        $this->bigtableClient->mutateRows(self::TABLE_NAME, $this->entries, $this->options)
+            ->shouldBeCalled()
+            ->willThrow($apiException);
+        try {
+            $this->table->mutateRows($this->rowMutations);
+            $this->fail('Expected an Exception, but no exception was thrown.');
+        } catch (BigtableDataOperationException $ex) {
+            $this->assertEquals($apiException->getCode(), $ex->getCode());
+            $this->assertEquals($exMessage, $ex->getMessage());
+            $this->assertSame(
+                $apiException->getErrorInfoMetadata(),
+                array_intersect_key(
+                    $ex->getMetadata(),
+                    $apiException->getErrorInfoMetadata()
+                )
```

Comment:
This is a bit of a nit, but I might prefer this to be a little more explicit, since both of these could be returning empty arrays and this would be "true". For instance, you could set the `metadata` above to a local variable, and use _that_ in the assertion.

Reply:
Fixed!